### DISCUSSION TITLE
OpenGL

### DISCUSSION THREAD

**Discussion by Gonzalomario** [skyline-emu/skyline#896] [Ideas]
why didn't they add an option to switch to OpenGL

**Comment by JERD001** [comment]
OpenGL isn't an option and will never be, Vulkan is the best way to go.

**Reply by davyson123** [reply]
But there's a weak vulcan (GPU) buddy

**Reply by JERD001** [reply]
Those weak GPU most likely don't have the features to actually run the emulator in the first place. Old PowerVR, some old Mali GPUs, and Adreno below 600 series aren't enough for the emulator.

**Reply by davyson123** [reply]
my and a mali g52 mc2 will it run buddy?

**Reply by therealfuranku** [reply]
I also had the same issue as well, my phone has PowerVR GE[8320](tel:8320) and it has Vulkan Support. idk why it doesn't work and keeps crashing.

"Old PowerVR, some old Mali GPUs, and Adreno below 600 series aren't enough for the emulator."

- Except for Mali GPUs, PPSSPP works on lesser Adreno and PowerVR GPUs. Because the emulator itself supports outdated GPUs, Vulkan works without a hitch. AetherSX2 is the same; I tried to use Vulkan with this emulator and it crashed. However, they introduced support for older GPUs, which works perfectly. Worst of all, I can't play several PS2 games because my phone isn't designed to run emulated 3D games other than PPSSPP, ePSXe, and the N64 emulator.

However, Skyline should have this, however keep in mind that while 3D games will not work, 2D games such as Katana Zero, Hyper Light Drifter, and Visual Novels such as DDLC Plus should work properly. Any 2D game created using Unity Engine should run smoothly at 30 fps.

**Comment by PixelyIon** [comment]
OpenGL ES is a legacy API that doesn't have the level of control over the host GPU we need for the Switch. It will eventually be discarded by vendors entirely and just implemented using a compatibility layer such as [`ANGLE`](https://github.com/google/angle) which implements OGLES atop Vulkan, AMD already does so on Android and other vendors will likely follow once they deem it pointless to have to constantly maintain their OGLES drivers.
